src/main/java/teammates/ui/webapi/DeleteFeedbackQuestionAction.java

```diff
@@ -16,25 +19,72 @@ AuthType getMinAuthLevel() {
     @Override
     void checkSpecificAccessControl() throws UnauthorizedAccessException {
         String feedbackQuestionId = getNonNullRequestParamValue(Const.ParamsNames.FEEDBACK_QUESTION_ID);
-        FeedbackQuestionAttributes questionAttributes = logic.getFeedbackQuestion(feedbackQuestionId);
+        UUID questionId;
+        FeedbackQuestionAttributes questionAttributes = null;
+        FeedbackQuestion question = null;
+        String courseId;
+
+        try {
+            questionId = getUuidRequestParamValue(Const.ParamsNames.FEEDBACK_QUESTION_ID);
+            question = sqlLogic.getFeedbackQuestion(questionId);
+        } catch (InvalidHttpParameterException e) {
+            questionAttributes = logic.getFeedbackQuestion(feedbackQuestionId);
+        }
 
-        if (questionAttributes == null) {
-            throw new UnauthorizedAccessException("Unknown question ID");
+        if (questionAttributes != null) {
+            courseId = questionAttributes.getCourseId();
+        } else {
+            if (question == null) {
+                throw new EntityNotFoundException("Unknown question id");
+            }
+            courseId = question.getCourseId();
         }
```

Comment:
just a preference, but i think it would be neater if this block of code could follow a similar logic as lines 63 to 68:

```
if (questionAttributes != null) {
            courseId = questionAttributes.getCourseId();
 } else if (question != null ) {
            courseId = question.getCourseId();
 } else {
     throw new EntityNotFoundException("Unknown question id");
}
```